sample-apps/go-sample-app/collection/config.go

```diff
@@ -0,0 +1,49 @@
+package collection
+
+import (
+	"github.com/spf13/viper"
+)
+
+// Config contains random based metrics; values inputed by configuration file or defaulted values
+type Config struct {
+	Host                    string
+	Port                    string
+	TimeInterval            int64
+	TimeAliveIncrementer    int64
+	TotalheapSizeUpperBound int64
+	ThreadsActiveUpperBound int64
+	CpuUsageUpperBound      int64
+}
+
+// GetConfiguration returns a configured Config struct with the precedence; Default Values < Configuration File.
+func GetConfiguration() *Config {
+	viper.SetConfigFile("config.yaml")
+	viper.ReadInConfig()
+	// Default values
```

Comment:
Do defaults need to be set before reading in the config file or only before reading values?